satellite/repair/checker/checker.go

```diff
@@ -271,19 +271,22 @@ func (obs *checkerObserver) RemoteSegment(ctx context.Context, path storj.Path,
 		}
 
 		// delete always returns nil when something was deleted and also when element didn't exists
-		err = obs.irrdb.Delete(ctx, []byte(path))
+		err = obs.irrdb.Delete(ctx, []byte(path.Raw))
 		if err != nil {
 			obs.log.Error("error deleting entry from irreparable db", zap.Error(err))
 			return nil
 		}
 	} else if numHealthy < redundancy.MinReq && numHealthy < redundancy.RepairThreshold {
-		pathElements := storj.SplitPath(path)
+		// TODO: see whether this can be handled with metainfo.ScopedPath
+		pathElements := storj.SplitPath(path.Raw)
```

Comment:
I saw that you put `SegmentPath` onto `ScopedPath` in previous commits. Seems like it would be nice to use that here. What's the reason that you decided not to do that?

Reply:
The reason is that here, it's not really a "segment path", but something else. I also suspect that this code is actually wrong in the first place.

Also, the concept of "segment path" isn't well defined... i.e. should it contain the segment number or not?... so yeah, opted out of implementing it for now.